src/components/HomePage/RepositoryList/RepositoryListItem/index.tsx

```diff
@@ -6,8 +6,9 @@ import "./RepositoryListItem.css";
 import { useDispatch } from "react-redux";
 import { addRepository, removeRepository } from "../../../../redux";
 
-const RepositoryListItem = ({ repository }: RepositoryListItemProps) => {
-  const [favorite, setFavorite] = useState(false);
+const RepositoryListItem = ({ repository, fav }: RepositoryListItemProps) => {
```

Comment:
```suggestion
const RepositoryListItem = ({ repository, isFavorite }: RepositoryListItemProps) => {
```

You don't need to use state here, your parent can pass this flag